src/main/java/fr/esgi/al5_2/Tayarim/services/AuthService.java

```diff
@@ -3,87 +3,137 @@
 
         return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
+                id,
+                token,
+                isAdmin
         );
-
     }
 
-    public AuthLoginResponseDTO authProprietaire(@NonNull String token){
+    public AuthLoginResponseDTO auth(@NonNull String token) {
+
+        Entry<Long, Boolean> entry = verifyToken(token, false);
 
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
-        
         return new AuthLoginResponseDTO(
-                proprietaireDTO.getId(),
-                token
+                entry.getKey(),
+                token,
+                entry.getValue()
         );
 
     }
 
-    public void logoutProprietaire(@NonNull String token){
+    public void logout(@NonNull String token) {
 
-        ProprietaireDTO proprietaireDTO = verifyToken(token);
+        Entry<Long, Boolean> entry = verifyToken(token, false);
 
-        tokenCacheService.addToCache(proprietaireDTO.getId(), UUID.randomUUID().toString());
+        tokenCacheService.addToCache(entry.getKey(), UUID.randomUUID().toString());
 
     }
 
-    public ProprietaireDTO verifyToken(@NonNull String token){
+    public Entry<Long, Boolean> verifyToken(@NonNull String token, boolean shouldBeAdmin) {
         ProprietaireDTO proprietaireDTO;
-        try{
-            proprietaireDTO = proprietaireService.getProprietaireByEmail(jwtHelper.extractEmail(token));
-        } catch (ProprietaireNotFoundException ex){
+        AdministrateurDTO administrateurDTO;
+        Long id;
+        String email;
+        boolean isAdmin = jwtHelper.extractAdmin(token);
+        if (shouldBeAdmin && !isAdmin) {
             throw new TokenExpireOrInvalidException();
         }
 
-        String uuid = tokenCacheService.getFromCache(proprietaireDTO.getId());
-        if(uuid == null){
+        if (isAdmin) {
+            try {
+                administrateurDTO = administrateurService.getAdministrateurByEmail(jwtHelper.extractEmail(token));
+                id = administrateurDTO.getId();
+                email = administrateurDTO.getEmail();
+            } catch (AdministrateurNotFoundException ex) {
+                throw new TokenExpireOrInvalidException();
+            }
+        } else {
+            try {
+                proprietaireDTO = proprietaireService.getProprietaireByEmail(jwtHelper.extractEmail(token));
+                id = proprietaireDTO.getId();
+                email = proprietaireDTO.getEmail();
+            } catch (ProprietaireNotFoundException ex) {
+                throw new TokenExpireOrInvalidException();
+            }
+        }
+
+        String uuid = tokenCacheService.getFromCache(id);
+        if (uuid == null) {
             throw new TokenExpireOrInvalidException();
         }
 
-        if(!jwtHelper.validateToken(token, proprietaireDTO.getEmail(), uuid)){
+        if (!jwtHelper.validateToken(token, email, uuid)) {
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.indentation.IndentationCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'if' has incorrect indentation level 8, expected level should be 4.